Runtime/PaletteFolder.cs

```diff
@@ -16,14 +16,17 @@ public class PaletteFolder
         [SerializeReference] private List<PaletteEntry> entries = new List<PaletteEntry>();
         public List<PaletteEntry> Entries => entries;
 
+        [SerializeField] private string guid;
+        
         public void Initialize(string name)
         {
             this.name = name;
+            guid = Guid.NewGuid().ToString();
         }
 
         public override string ToString()
         {
-            return $"{GetType().Name}({Name})";
+            return $"[{guid}] {GetType().Name}({Name})";;
```

Comment:
The type and the name of the folder are the most important for debugging, better if GUID is at the back
```suggestion
            return $"{GetType().Name}({Name}, {guid})";
```